diff --git a/scripts/mkimage.sh b/scripts/mkimage.sh
index 82c53d91..3476c759 100755
--- a/scripts/mkimage.sh
+++ b/scripts/mkimage.sh
@@ -20,7 +20,6 @@ git=$(command -v git) || git=true
 
 # deduce aports directory
 [ -n "$APORTS" ] || APORTS=$(realpath "$scriptdir/../")
-[ -e "$APORTS/main/build-base" ] || die "Unable to deduce aports base checkout"
 
 # echo '-dirty' if git is not clean
 git_dirty() {
